apps/antalmanac/src/stores/scheduleHelpers.ts

```diff
@@ -167,23 +167,20 @@ export function getColorForNewSection(newSection: ScheduleCourse, sectionsInSche
         (course) => course.section.sectionType === newSection.section.sectionType
     );
 
-    const usedColors = new Set(sectionsInSchedule.map((course) => course.section.color));
+    const usedColors = sectionsInSchedule.map((course) => course.section.color);
+    const lastDefaultColor = usedColors.findLast((materialColor) =>
+        (defaultColors as string[]).includes(materialColor)
+    ) as unknown as (typeof defaultColors)[number];
 
     // If the same sectionType exists, return that color
     if (existingSectionsType.length > 0) return existingSectionsType[0].section.color;
 
     // If the same courseTitle exists, but not the same sectionType, return a close color
-    if (existingSections.length > 0) return generateCloseColor(existingSections[0].section.color, usedColors);
+    if (existingSections.length > 0) return generateCloseColor(existingSections[0].section.color, new Set(usedColors));
 
-    // If there are no existing sections with the same course title, generate a new color. If we run out of unique colors, return the next color after the last one in use, looping after reaching the end.
+    // If there are no existing sections with the same course title, generate a new color. If we run out of unique colors, return the next color up after the last default color in use, looping after reaching the end.
     return (
-        defaultColors.find((materialColor) => !usedColors.has(materialColor)) ||
-        defaultColors[
-            (defaultColors.indexOf(
-                sectionsInSchedule[sectionsInSchedule.length - 1].section.color as (typeof defaultColors)[number]
-            ) +
-                1) %
-                7
-        ]
+        defaultColors.find((materialColor) => !usedColors.includes(materialColor)) ||
+        defaultColors[(defaultColors.indexOf(lastDefaultColor) + 1) % 7]
```

Comment:
I think that instead of hardcoding the length of `defaultColors`, it is better to use the `defaultColors.length` directly or as a `const`, in case more default colors are added, the code will automatically adapt.
Otherwise, looks good.

Reply:
That's a good point, I'll fix that.

Reply:
Just pushed a commit that fixes this issue.